### ISSUE TITLE
keras 1.0.4 sharing batchnormalization layer across different data flows.

### ISSUE THREAD

**Issue by isaacgerg** [fchollet/keras#3903]
```
    input_tensor = keras.layers.Input(shape=(3, 200, 200))
    resnetModel = keras.applications.resnet50.ResNet50(include_top=False, weights='imagenet', input_tensor=input_tensor)

    for k in resnetModel.layers:
        print(k)

    #split = 162

    #for k in resnetModel.layers[:split]:
    #    k.trainable = False

    model = Sequential()                
    model.add(AveragePooling2D(pool_size=(2,2), input_shape=(1, 400, 400))) # Downsample input data
    model.add(Flatten())
    model.add(keras.layers.RepeatVector(3))
    model.add(Reshape((3, 200, 200)))                          
    model.add(resnetModel)
```

This code gives me the following and am unsure why.  Any ideas?
builtins.Exception: You are attempting to share a same `BatchNormalization` layer across different data flows. This is not possible. You should use `mode=2` in `BatchNormalization`, which has a similar behavior but is shareable (see docs for a description of the behavior).

**Comment by dolaameng** [comment]
I guess the way for ResNet to be piped with other models (without creating another data flow) is to pass the output of a model to its `input_tensor` parameter. Something like this
```python
x = Input(shape=(1, 400, 400))
y = AveragePooling2D(pool_size=(2, 2))(x)
y = Flatten()(y)
y = RepeatVector(3)(y)
y = Reshape((3, 200, 200), )(y)

resnet = resnet50.ResNet50(include_top=False, weights='imagenet', 
                           input_tensor=y)

model = Model(input=x, output=resnet.output)
```
But it got an `Graph disconnected` error because `y` is a Keras tensor (K.is_keras_tensor(y)=True) and is from an non-input layer. I also want to know how to do this.

**Comment by isaacgerg** [comment]
I tried the same thing and also got the graph disconnected issue.
